src/shared/components/oncoprint/tabularDownload.ts

```diff
@@ -104,49 +104,122 @@ export function getTabularDownloadData(
                     : geneticTrackDatum.patient!;
-            oncoprintData.CNA[currentGeneName][id] = '';
-            oncoprintData.MUTATIONS[currentGeneName][id] = '';
-            oncoprintData.MRNA[currentGeneName][id] = '';
-            oncoprintData.PROTEIN[currentGeneName][id] = '';
-            oncoprintData.STRUCTURAL_VARIANT[currentGeneName][id] = '';
+
+            // Only initialize each alteration for sample/patient if it doesn't exist
+            // That can happen when we have multiple OncoPrint tracks of the same gene (using OQL)
+            if (oncoprintData.CNA[currentGeneName][id] == undefined) {
+                oncoprintData.CNA[currentGeneName][id] = '';
+            }
+            if (oncoprintData.MUTATIONS[currentGeneName][id] == undefined) {
+                oncoprintData.MUTATIONS[currentGeneName][id] = '';
+            }
+            if (oncoprintData.MRNA[currentGeneName][id] == undefined) {
+                oncoprintData.MRNA[currentGeneName][id] = '';
+            }
+            if (oncoprintData.PROTEIN[currentGeneName][id] == undefined) {
+                oncoprintData.PROTEIN[currentGeneName][id] = '';
+            }
+            if (
+                oncoprintData.STRUCTURAL_VARIANT[currentGeneName][id] ==
+                undefined
+            ) {
+                oncoprintData.STRUCTURAL_VARIANT[currentGeneName][id] = '';
+            }
+
+            // Check if the sample/patient is initialized, and combine both results in case they are different
             if (geneticTrackDatum.disp_cna !== undefined) {
-                oncoprintData.CNA[currentGeneName][id] = cnaMap[
-                    geneticTrackDatum.disp_cna
-                ]
+                var current_value = cnaMap[geneticTrackDatum.disp_cna]
                     ? cnaMap[geneticTrackDatum.disp_cna]
                     : geneticTrackDatum.disp_cna;
+
+                if (
+                    oncoprintData.CNA[currentGeneName][id] == '' ||
+                    oncoprintData.CNA[currentGeneName][id] == current_value
+                ) {
+                    oncoprintData.CNA[currentGeneName][id] = current_value;
+                } else {
+                    oncoprintData.CNA[currentGeneName][id] = oncoprintData.CNA[
+                        currentGeneName
+                    ][id].concat(', ', current_value);
+                }
             }
             if (geneticTrackDatum.disp_structuralVariant !== undefined) {
-                oncoprintData.STRUCTURAL_VARIANT[currentGeneName][
-                    id
-                ] = structuralVariantMap[
+                var current_value = structuralVariantMap[
                     geneticTrackDatum.disp_structuralVariant + ''
                 ]
                     ? structuralVariantMap[
                           geneticTrackDatum.disp_structuralVariant + ''
                       ]
                     : geneticTrackDatum.disp_structuralVariant;
+                if (
```

Comment:
not sure i understand this if statement.  could use. comment

Reply:
I've reviewed it. There were some mistakes in the code, I've just fixed them.

We might be able to simplify this since we're doing the same for the different alteration types (cna, mutations, mRNA, protein, and structural variants) using a for loop, but we would probably need a dictionary since the parts they change between alterations are not exactly the same (e.g. disp_cna and CNA in CNA, disp_structuralVariant and STRUCTURAL_VARIANT for structural variants, etc.). Do you think it's a good idea?

Reply:
I agree that these look exactly the same and this should be abstracted.     not sure i see where the dictionary is necessary

Reply:
I've abstracted the code. Now you can see what I meant with the dictionary